### PR TITLE
Fix misformatted filename for grid engine and slurm with CWL (resolves #1261)

### LINKED ISSUE #1261: Failure when running CWL on slurm
I'm having trouble getting a simple CWL definition to execute on slurm.  Can someone help me understand what I'm messing up?

The error is ` IOError: [Errno 2] No such file or directory: 'file:/working/cwarth/cft/toil/jobStore/results.txt'`

This looks very similar to #1108 when using the GridEngine batch system, but the workaround of using ` --jobStore ./jobStore` does not help.
&nbsp;
      * [`cwltoil` fails when run with `--batchSystem slurm`](#cwltoil_w_slurm)
      * [version info](#version)
      * [`hello.cwl` and `hello.yml` test scripts](#sources)

Working Examples:
      * [Toil script without CWL works with Slurm,](#toil_wo_cwl)
      * [`cwltool` can run CWL without Slurm](#cwltool_wo_slurm)
      * [`cwltoil` can run CWL without Slurm](#cwltoil_wo_slurm)

<h2 id="cwltoil_w_slurm">`cwltoil` fails when run with `--batchSystem slurm`</h2>

```
$ cwltoil --logDebug --batchSystem slurm --disableCaching --jobStore ./jobStore --no-container hello.cwl hello.yml                                                                                                                                                                                                             
stoat 2016-11-01 10:59:50,870 MainThread INFO toil.lib.bioio: Root logger is at level 'DEBUG', 'toil' logger at level 'DEBUG'.
stoat 2016-11-01 10:59:54,277 MainThread INFO toil.lib.bioio: Root logger is at level 'DEBUG', 'toil' logger at level 'DEBUG'.
stoat 2016-11-01 10:59:54,718 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/working/cwarth/cft/toil/jobStore'.
stoat 2016-11-01 10:59:54,844 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '3c805a37-44aa-4c4d-84f0-38faa5370313'
stoat 2016-11-01 10:59:55,293 MainThread INFO toil.common: Using the slurm batch system
stoat 2016-11-01 10:59:55,293 MainThread DEBUG toil.batchSystems.slurm: slurm results file is file:/working/cwarth/cft/toil/jobStore/results.txt
Traceback (most recent call last):
  File "/working/cwarth/cft/toil/venv/bin/cwltoil", line 11, in <module>
    load_entry_point('toil', 'console_scripts', 'cwltoil')()
  File "/working/cwarth/cft/toil/venv/src/toil/src/toil/cwl/cwltoil.py", line 665, in main
    outobj = toil.start(wf1)
  File "/working/cwarth/cft/toil/venv/src/toil/src/toil/common.py", line 591, in start
    self._batchSystem = self.createBatchSystem(self.config)
  File "/working/cwarth/cft/toil/venv/src/toil/src/toil/common.py", line 758, in createBatchSystem
    return batchSystemClass(**kwargs)
  File "/working/cwarth/cft/toil/venv/src/toil/src/toil/batchSystems/slurm.py", line 282, in __init__
    self.slurmResultsFileHandle = open(self.slurmResultsFile, 'w')
IOError: [Errno 2] No such file or directory: 'file:/working/cwarth/cft/toil/jobStore/results.txt'

```

<h2 id="version">version info</h2>

```
$ python --version
Python 2.7.6
(venv)cwarth@stoat ~/src/matsen/cft/toil (master) 
$ pip freeze | grep toil
toil==3.5.0a1
```

`toil` is installed directly from github with,

```
pip install -e git+https://github.com/BD2KGenomics/toil#egg=toil[cwl]
```

<h2 id="sources">`hello.cwl` and `hello.yml` test scripts</h2>

```
$ cat hello.cwl
cwlVersion: v1.0
class: CommandLineTool
baseCommand: echo
inputs:
  message:
    type: string
    inputBinding:
      position: 1
outputs: 
  output:
    type: stdout
stdout: $(inputs.message).ext
```

```
$ cat hello.yml
message: foo.bar
```


<h2 id="toil_wo_cwl">Toil script without CWL works with Slurm,</h2>

```
$ python HelloWorld.py --batchSystem slurm --disableCaching --logDebug ./jobStore
$ python HelloWorld.py  --logDebug --batchSystem slurm --disableCaching --jobStore ./jobStore --logFile logfile --no-container 
stoat 2016-11-01 11:09:49,520 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
stoat 2016-11-01 11:09:49,521 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
stoat 2016-11-01 11:09:49,972 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/working/cwarth/cft/toil/toilWorkflow'.
stoat 2016-11-01 11:09:50,021 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '9e858ae4-6ec4-45c2-bc48-285d0db66efa'
stoat 2016-11-01 11:09:50,334 MainThread INFO toil.common: Using the single machine batch system
stoat 2016-11-01 11:09:50,334 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxCores to CPU count of system (16).
stoat 2016-11-01 11:09:50,334 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxMemory to physically available memory (67378266112).
stoat 2016-11-01 11:09:50,335 MainThread INFO toil.common: Created the workflow directory at /tmp/toil-9e858ae4-6ec4-45c2-bc48-285d0db66efa
stoat 2016-11-01 11:09:50,335 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxDisk to physically available disk (4560371712).
stoat 2016-11-01 11:09:50,336 MainThread INFO toil.batchSystems.singleMachine: Setting up the thread pool with 160 workers, given a minimum CPU fraction of 0.100000 and a maximum CPU value of 16.
stoat 2016-11-01 11:09:50,390 MainThread INFO toil.common: No user script to hot-deploy.
stoat 2016-11-01 11:09:50,421 MainThread INFO toil.common: Written the environment for the jobs to the environment file
stoat 2016-11-01 11:09:50,421 MainThread INFO toil.common: Caching all jobs in job store
stoat 2016-11-01 11:09:50,453 MainThread INFO toil.common: 0 jobs downloaded.
stoat 2016-11-01 11:09:50,810 MainThread INFO toil.realtimeLogger: Real-time logging disabled
stoat 2016-11-01 11:09:51,185 MainThread INFO toil.leader: (Re)building internal scheduler state
stoat 2016-11-01 11:09:51,186 MainThread INFO toil.leader: Checked batch system has no running jobs and no updated jobs
stoat 2016-11-01 11:09:51,187 MainThread INFO toil.leader: Found 1 jobs to start and 0 jobs with successors to run
stoat 2016-11-01 11:09:51,187 MainThread INFO toil.leader: Starting the main loop
stoat 2016-11-01 11:09:51,229 Thread-1 INFO toil.batchSystems.singleMachine: Executing command: '/working/cwarth/cft/toil/venv/bin/_toil_worker file:/working/cwarth/cft/toil/toilWorkflow q/x/jobJt2wuN'.
stoat 2016-11-01 11:09:59,293 MainThread INFO toil.leader: No jobs left to run so exiting.
stoat 2016-11-01 11:09:59,294 MainThread INFO toil.leader: Finished the main loop
stoat 2016-11-01 11:09:59,294 MainThread INFO toil.leader: Waiting for stats and logging collator thread to finish ...
stoat 2016-11-01 11:09:59,645 Thread-162 INFO toil.leader: Got message from job at time 11-01-2016 11:09:59: Job q/x/jobJt2wuN/g/tmpx6jOcL.tmp used 0.00% (0.0 B [0B] used, 3.0 GB [3221225472B] requested) at the end of its run.
stoat 2016-11-01 11:09:59,833 MainThread INFO toil.leader: ... finished collating stats and logs. Took 0.538393974304 seconds
stoat 2016-11-01 11:09:59,833 MainThread INFO toil.leader: Waiting for service manager thread to finish ...
stoat 2016-11-01 11:10:00,188 MainThread INFO toil.leader: ... finished shutting down the service manager. Took 0.355026006699 seconds
stoat 2016-11-01 11:10:00,188 MainThread INFO toil.leader: Finished toil run successfully
stoat 2016-11-01 11:10:00,276 MainThread INFO toil.common: Attempting to delete the job store
stoat 2016-11-01 11:10:00,936 MainThread INFO toil.common: Successfully deleted the job store
Hello, world!, here's a message: woot
```

<h2 id="cwltool_wo_slurm">`cwltool` can run CWL without Slurm</h2>

```
$ cwltool --no-container hello.cwl  hello.yml                                                                                                                                                                                                                                                                                                                         
/working/cwarth/cft/toil/venv/bin/cwltool 1.0.20160714182449
[job hello.cwl] /tmp/tmpaiOU7c$ echo \
    foo.bar > /tmp/tmpaiOU7c/results.txt
Final process status is success
{
    "output": {
        "checksum": "sha1$f9299c817b423a41aae4bbce4b17f5edcb9c9282", 
        "basename": "results.txt", 
        "location": "file:///working/cwarth/cft/toil/results.txt", 
        "path": "/working/cwarth/cft/toil/results.txt", 
        "class": "File", 
        "size": 8
    }
}
```

<h2 id="cwltoil_wo_slurm">`cwltoil` can run CWL without Slurm</h2>

```
$ cwltoil --no-container hello.cwl hello.yml                                                                                                                                                                                                                                                                                                                              
stoat 2016-11-01 11:06:36,753 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
stoat 2016-11-01 11:06:39,447 MainThread INFO toil.lib.bioio: Root logger is at level 'INFO', 'toil' logger at level 'INFO'.
stoat 2016-11-01 11:06:39,945 MainThread INFO toil.jobStores.fileJobStore: Path to job store directory is '/tmp/tmpuMpTs9'.
stoat 2016-11-01 11:06:39,946 MainThread INFO toil.jobStores.abstractJobStore: The workflow ID is: '2dbb34d1-ce9f-4a64-99dd-1955980412f8'
stoat 2016-11-01 11:06:40,483 MainThread INFO toil.common: Using the single machine batch system
stoat 2016-11-01 11:06:40,483 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxCores to CPU count of system (16).
stoat 2016-11-01 11:06:40,483 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxMemory to physically available memory (67378266112).
stoat 2016-11-01 11:06:40,484 MainThread INFO toil.common: Created the workflow directory at /tmp/toil-2dbb34d1-ce9f-4a64-99dd-1955980412f8
stoat 2016-11-01 11:06:40,484 MainThread WARNING toil.batchSystems.singleMachine: Limiting maxDisk to physically available disk (4560850944).
stoat 2016-11-01 11:06:40,484 MainThread INFO toil.batchSystems.singleMachine: Setting up the thread pool with 160 workers, given a minimum CPU fraction of 0.100000 and a maximum CPU value of 16.
stoat 2016-11-01 11:06:40,565 MainThread INFO toil.common: User script ModuleDescriptor(dirPath='/working/cwarth/cft/toil/venv/src/toil/src', name='toil.cwl.cwltoil', fromVirtualEnv=True) belongs to Toil. No need to hot-deploy it.
stoat 2016-11-01 11:06:40,565 MainThread INFO toil.common: No user script to hot-deploy.
stoat 2016-11-01 11:06:40,566 MainThread INFO toil.common: Written the environment for the jobs to the environment file
stoat 2016-11-01 11:06:40,566 MainThread INFO toil.common: Caching all jobs in job store
stoat 2016-11-01 11:06:40,566 MainThread INFO toil.common: 0 jobs downloaded.
stoat 2016-11-01 11:06:40,571 MainThread INFO toil.realtimeLogger: Real-time logging disabled
stoat 2016-11-01 11:06:41,021 MainThread INFO toil.leader: (Re)building internal scheduler state
stoat 2016-11-01 11:06:41,022 MainThread INFO toil.leader: Checked batch system has no running jobs and no updated jobs
stoat 2016-11-01 11:06:41,023 MainThread INFO toil.leader: Found 1 jobs to start and 0 jobs with successors to run
stoat 2016-11-01 11:06:41,024 MainThread INFO toil.leader: Starting the main loop
stoat 2016-11-01 11:06:41,075 Thread-1 INFO toil.batchSystems.singleMachine: Executing command: '/working/cwarth/cft/toil/venv/bin/_toil_worker file:/tmp/tmpuMpTs9 L/S/job0TZ6_5'.
stoat 2016-11-01 11:06:51,240 MainThread INFO toil.leader: No jobs left to run so exiting.
stoat 2016-11-01 11:06:51,240 MainThread INFO toil.leader: Finished the main loop
stoat 2016-11-01 11:06:51,241 MainThread INFO toil.leader: Waiting for stats and logging collator thread to finish ...
stoat 2016-11-01 11:06:51,541 Thread-162 INFO toil.leader: Got message from job at time 11-01-2016 11:06:51: Job L/S/job0TZ6_5/g/tmpUbWKa4.tmp used 0.00% (0.0 B [0B] used, 2.0 GB [2147483648B] requested) at the end of its run.
stoat 2016-11-01 11:06:51,541 Thread-162 INFO toil.leader: Got message from job at time 11-01-2016 11:06:51: Successfully deleted local copies of file with ID 'R/L/tmpSVgIYD.tmp'.
stoat 2016-11-01 11:06:51,542 MainThread INFO toil.leader: ... finished collating stats and logs. Took 0.301622867584 seconds
stoat 2016-11-01 11:06:51,543 MainThread INFO toil.leader: Waiting for service manager thread to finish ...
stoat 2016-11-01 11:06:52,026 MainThread INFO toil.leader: ... finished shutting down the service manager. Took 0.4829621315 seconds
stoat 2016-11-01 11:06:52,026 MainThread INFO toil.leader: Finished toil run successfully
stoat 2016-11-01 11:06:52,430 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.azureJobStore'. You may want to try reinstalling Toil with additional extras.
stoat 2016-11-01 11:06:52,755 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.googleJobStore'. You may want to try reinstalling Toil with additional extras.
stoat 2016-11-01 11:06:53,166 MainThread INFO toil.jobStores.abstractJobStore: Unable to import 'toil.jobStores.aws.jobStore'. You may want to try reinstalling Toil with additional extras.
{
    "output": {
        "checksum": "sha1$f9299c817b423a41aae4bbce4b17f5edcb9c9282", 
        "basename": "results.txt", 
        "location": "file:///tmp/tmpoMG_XW/results.txt", 
        "path": "/working/cwarth/cft/toil/results_2.txt", 
        "class": "File", 
        "size": 8
    }
stoat 2016-11-01 11:06:53,188 MainThread INFO toil.common: Attempting to delete the job store
stoat 2016-11-01 11:06:53,190 MainThread INFO toil.common: Successfully deleted the job store
}
```

### PR DESCRIPTION
**Pull request by jessebrennan**
First commit updates tests which should cause failure. Second commit should fix failure.

### CODE REVIEW

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 1]
```diff
@@ -262,6 +262,23 @@ def testJobConcurrency(self):
...
+        """
+
+        def _createDummyConfig(self):
+            config = super(hidden.AbstractGridEngineBatchSystemTest, self)._createDummyConfig()
+            # can't use _getTestJobStorePath since that method removes the directory
+            config.jobStore = 'file:' + self._createTempDir('jobStore')
+            return config
+
+        def testResultFile(self):
+            """tests that the result file name is formatted properly"""
```
Comment (hannes-ucsc) [inline src/toil/test/batchSystems/batchSystemTest.py]: Be consistent with the formatting of other docstrings in Toil.

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 2]
```diff
@@ -262,6 +262,23 @@ def testJobConcurrency(self):
...
+            file_name = BatchSystemSupport._getResultsFileName(self._createDummyConfig().jobStore)
```
Comment (hannes-ucsc) [inline src/toil/test/batchSystems/batchSystemTest.py]: Toil uses camel case for variable names.

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 3]
```diff
@@ -262,6 +262,23 @@ def testJobConcurrency(self):
...
+            self.assertTrue(file_name.startswith('file:'))
```
Comment (hannes-ucsc) [inline src/toil/test/batchSystems/batchSystemTest.py]: There is more you can assert and more cases you can test.

File: src/toil/test/batchSystems/batchSystemTest.py [hunk 4]
```diff
@@ -288,7 +288,8 @@ def testResultFile(self):
             """
             Tests that the result file name is formatted properly
             """
-            fileName = BatchSystemSupport._getResultsFileName(self._createConfig().jobStore)
+            # noinspection PyUnresolvedReferences
```
Comment (jessebrennan) [inline src/toil/test/batchSystems/batchSystemTest.py]: could also use `assert isinstance(self.batchSystem, BatchSystemSupport` to remove pyCharm warning. Preference?

**Comment by hannes-ucsc** [comment]
The second commit should end in `(resolves #1261)`

One of the commits in a PR should carry the `resolves` or `fixes` qualifiers. Merging the PR will then automatically close the referenced issue.

The commit messages should be more general. The way they are written now lacks context. Prefer generality over specifics if the specifics don't make sense without context. Also try to keep commit titles (the first line in the message) no longer than 76 characters.

**Comment by jessebrennan** [comment]
@hannes-ucsc I had to redo and add the extra commits above because the tests weren't failing for exactly the right reason and because a test that wasn't supposed to fail failed. That has all been fixed and the most recent commit should resolve the issue. 

I will work on creating more legible commit messages. Is it worth going back to fix the ones I've pushed already for this branch?

**Comment by hannes-ucsc** [comment]
> Is it worth going back to fix the ones I've pushed already for this branch?
 yes

**Comment by hannes-ucsc** [comment]
IT failure is another PyPI outage.

**Comment by hannes-ucsc** [comment]
Remember that I want one commit with a test that exposes the problem, and another commit that fixes the problem. Each commit should be pushed separately such that each commit triggers a build. Git gives you all the tools to get there. Come find me in my office if you need guidance.

**Comment by hannes-ucsc** [comment]
In the future, don't don't close the PR. Just force push to the same branch. If you do close a PR and continue it in another PR, cross-reference the PRs via comments.

**Comment by jessebrennan** [comment]
Continued https://github.com/BD2KGenomics/toil/pull/1355

**Pull request closed without merging**